### ISSUE TITLE
Don't build happy or alex with GHCJS

### ISSUE THREAD

**Issue by basvandijk** [NixOS/nixpkgs#7264]
Haskell packages like `pretty-show` or `language-python` need `happy` or `alex` as build tools and therefor list them in `buildTools`. When building these packages using GHCJS, nix tries to build `happy` or `alex` using GHCJS which fails. I currently override the packages to always use a happy or alex compiled with GHC. This is my override function:  

```nix
pkgs : {
  overrides = self: super:
    with (import <nixpkgs/pkgs/development/haskell-modules/lib.nix> {inherit pkgs;});
    let adaptBuildTools = drv: f: overrideCabal drv (drv: { buildTools = f drv.buildTools; });
    in {

    # Overrides
    "pretty-show"     = adaptBuildTools super.pretty-show     (_buildTools: [pkgs.haskellngPackages.happy]);
    "language-python" = adaptBuildTools super.language-python (_buildTools: [pkgs.haskellngPackages.happy
                                                                             pkgs.haskellngPackages.alex]);
  };
}
```

It would be nicer if `happy` and `alex` (and other Haskell-based build tools) will always be build using GHC.

@peti any idea how to best accomplish this?

**Comment by peti** [comment]
Maybe `configuration-ghcjs.nix` should re-define attributes like `happy` and `alex` to refer to those from the GHC 7.10.1 package set?

Cc: @cstrahan

**Comment by cstrahan** [comment]
Fixed in ed30c8ba409093.
